pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -86,6 +86,10 @@ func (r *Cluster) ValidateUpdate(old runtime.Object) error {
 		return apierrors.NewBadRequest(fmt.Sprintf("expected a Cluster but got a %T", old))
 	}
 
+	if r.IsSelfManaged() && !r.IsReconcilePaused() && features.IsActive(features.FullLifecycleAPI()) {
+		return apierrors.NewBadRequest(fmt.Sprintf("upgrading self managed clusters is not supported: %s", r.Name))
```

Comment:
We need to update this for the `Create` function too right?

Reply:
yes, for create the logic will be updated to return the requests, and then validate the spec, similar to this logic. Will be doing it as a part of a separate PR though. This is for Update validations.

Reply:
Also, what difference does it make checking for the envar in this case? We don't want to support it when the feature is enabled but when not, workload cluster is supported? Should it be separate conditions then?